low-level/files/opt/seagate/sspl/bin/sspl_setup

```diff
@@ -1,97 +1,234 @@
-    -r  Role to be configured on the current node
-config options:
-    -f  Force reinitialization. Do not prompt
-    -r  Role to be configured on the current node
-post_install options:
-    -p Product to be configured
-reset options:
-    -p product to be configured
-EOF
-    exit 1
-}
-
-cmd=
-[ $# -ge 1 ] && cmd=$1 && shift 1
-
-case $cmd in
-    post_install )
-        $SCRIPT_DIR/sspl_post_install $*
-        exit $?
-        ;;
-
-    init )
-        $SCRIPT_DIR/sspl_setup_init $*
-        exit $?
-        ;;
-
-    config )
-        $SCRIPT_DIR/sspl_config $*
-        exit $?
-        ;;
-
-    test )
-        $SCRIPT_DIR/sspl_test $*
-        exit $?
-        ;;
-
-    support_bundle )
-        $SCRIPT_DIR/sspl_bundle_generate $*
-        exit $?
-        ;;
-
-    manifest_support_bundle )
-        $SCRIPT_DIR/manifest_support_bundle $*
-        exit $?
-        ;;
-
-    reset )
-        $SCRIPT_DIR/sspl_reset $*
-        exit $?
-        ;;
-
-    check )
-        sudo python3 $SCRIPT_DIR/validate_consul_config.py
-        [ -f $SSPL_CONFIGURED ] && exit $?
-        logger -i -p local3.err "SSPL is not configured. Run provisioner scripts in $SSPL_BASE_DIR/bin."
-        exit 1
-        ;;
-
-    * )
-        usage
-        ;;
-esac
+import sys
+import errno
+import argparse
+import inspect
+import traceback
+import os
+import subprocess
+# Add the top level directories
+sys.path.insert(0, '/opt/seagate/cortx/sspl/low-level')
+from framework.base.sspl_constants import PRODUCT_FAMILY
+from files.opt.seagate.sspl.bin import validate_consul_config
+
+class Cmd:
+    """Setup Command"""
+
+    def __init__(self, args: dict):
+        self._args = args.args
+        self._script_dir = os.path.dirname(os.path.abspath(__file__))
+
+    @property
+    def args(self) -> str:
+        return self._args
+
+    @staticmethod
+    def usage(prog: str):
+        """Print usage instructions"""
+
+        sys.stderr.write(
+            f"{prog} [post_install[-p <LDR_R1>]|init [-dp] [-r <ssu|gw|cmu|vm>]\n|config [-f] [-r <ssu|gw|cmu|vm>]|test [self|sanity]|reset [hard -p <LDR_R1>|soft]]\n"
+            "init options:\n"
+            "\t -dp Create configured datapath\n"
+            "\t -r  Role to be configured on the current node\n"
+            "config options:\n"
+            "\t -f  Force reinitialization. Do not prompt\n"
+            "\t -r  Role to be configured on the current node"
+            "post_install options:\n"
+            "\t -p Product to be configured\n"
+            "reset options:\n"
+            "\t -p Product to be configured\n")
+
+    @staticmethod
+    def get_command(desc: str, argv: dict):
+        """Return the Command after parsing the command line."""
+
+        parser = argparse.ArgumentParser(desc)
+        subparsers = parser.add_subparsers()
+        cmds = inspect.getmembers(sys.modules[__name__])
+        cmds = [(x, y) for x, y in cmds
+            if x.endswith("Cmd") and x != "Cmd"]
```

Comment:
Does this condition apply really in sspl case. Did you test with prints for each argument like sspl_setup (post_install, config, init, etc) ?

Reply:
Ignoring this comment as already discussed in the call.